lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -117,7 +136,7 @@ data WalletState = WalletState
     { walletTip :: Read.ChainPoint
     , addresses :: !Address.AddressState
     , utxoHistory :: !UTxOHistory.UTxOHistory
-    , txHistory :: !TxHistory.TxHistory
+    , txHistory :: !TxHistory
```

Comment:
👍 This implies that we will eventually need to support `rollForward` and `rollBackward` on the `TxHistory` type, though. 🤓

Reply:
Yes, which means implementing a special split on time. The `ByCustomer` seems a bit more problematic in terms of complexity ...